ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -80,42 +81,44 @@
    */
   static class Proxies {
```

Comment:
We don't need to change Proxies since the same connection can be shared by multiple groups.

Reply:
@szetszwo 

I have rebased and can see most of the group id pass through code is not needed. 

However, I am still seeing a problem here about Proxies. 

First of all, I think we expect to set a group id in [DataStreamClient](https://github.com/apache/incubator-ratis/blob/master/ratis-client/src/main/java/org/apache/ratis/client/impl/DataStreamClientImpl.java#L48).  So each client will include this group id in the RaftClientRequest in the header.

However, in `NettyServerStreamRpc` we are doing `new Proxies(new PeerProxyMap<>(name, peer -> newClient(peer, properties)));`   So it leaves no place to pass in a group id to create DataStreamClient (e.g. `outs.add(map.getProxy(peer.getId()).stream());`)

That is why I am changing here to `Map<RaftPeerId, Map<RaftGroupId, DataStreamClient>> map`. By doing so, My intention is at least allow when creating a new DataStreamClient, we can set the group id.

Do you think whether this makes sense?

Reply:
I see the problem now.  Let's add a new method
```
  DataStreamOutput stream(RaftGroupId groupId);
```
to DataStreamApi; filed https://issues.apache.org/jira/browse/RATIS-1127

Reply:
@amaliujia Please rebase master to use DataStreamOutput stream(RaftGroupId groupId);